bin/cleaner/delete-buckets-for-non-existing-users.js

```diff
@@ -0,0 +1,191 @@
+#!/usr/bin/env node
+
+'use strict';
+
+const program = require('commander');
+const Storage = require('storj-service-storage-models');
+const mongoose = require('mongoose');
+const sqlDriver = require('mysql');
+
+const Config = require('../../lib/config');
+const NetworkMessageQueue = require('../../lib/server/queues/networkQueue');
+const { DELETING_FILE_MESSAGE } = require('../../lib/server/queues/messageTypes');
+const { iterateOverCursor, deleteBucketAndContents } = require('./database');
+
+// Example:
+// node bin/cleaner/delete-buckets-for-non-existing-users.js \
+// -d mariadb://root:example@localhost:3306/xCloud \
+// -u mongodb://admin:password@localhost:27017/__inxt-network \
+// -q amqp://admin:password@localhost:5672
+
+program.version('0.0.1')
+  .option('-c, --config <path_to_config_file>', 'path to the config file')
+  .option('-d, --dburl <db connection string>', 'sql db connection string')
+  .option('-u, --mongourl <mongo connection string>', 'mongo url connection string')
+  .option('-q, --queueurl <queue connection string>', 'queue url connection string')
+  .option('-s, --startFromBucket <bucket id to start script from>', 'bucket id to start script from')
+  .parse(process.argv);
+
+const DEFAULT_QUEUE_NAME = 'NETWORK_WORKER_TASKS_QUEUE';
+
+if (!process.env.NODE_ENV) {
+  throw new Error('NODE_ENV is not set');
+}
+
+if (!program.dburl) {
+  throw new Error('Add the DB connection string as -d or --dburl');
+}
+
+if (!program.queueurl) {
+  throw new Error('Add the Queue connection string as -q or --queueurl');
+}
+
+let idBucketBeingChecked;
+let deletedBuckets = 0;
+let checkedBuckets = 0;
+let startFromBucket;
+
+if (program.startFromBucket) {
+  startFromBucket = program.startFromBucket;
+}
+
+const sqlPool = sqlDriver.createPool(program.dburl);
+const networkQueue = new NetworkMessageQueue({
+  connection: {
+    url: program.queueurl,
+  },
+  exchange: {
+    name: 'exchangeName',
+    type: 'direct',
+  },
+  queue: {
+    name: program.queuename || DEFAULT_QUEUE_NAME,
+  },
+  routingKey: {
+    name: 'routingKeyName',
+  },
+});
+const config = new Config(process.env.NODE_ENV, program.config);
+const storage = new Storage(
+  program.mongourl || config.storage.mongoUrl,
+  config.storage.mongoOpts
+);
+
+const { Bucket: BucketModel, BucketEntry: BucketEntryModel, Frame:FrameModel, Pointer:PointerModel } = storage.models;
+
+const logStatus = () => {
+  console.log('Deleted buckets: ', deletedBuckets);
+  console.log('Checked buckets: ', checkedBuckets);
+  if (idBucketBeingChecked) {
+    console.log('Id of last bucket checked: ', idBucketBeingChecked);
+  }
+};
+
+const loggerInterval = setInterval(logStatus, 4000);
+
+const onEveryPointerDeleted = (pointer, nextPointer) => {
+  networkQueue.enqueueMessage({
+    type: DELETING_FILE_MESSAGE,
+    payload: { hash: pointer.hash }
+  }, (err) => {
+    if (err) {
+      return nextPointer(err);
+    }
+    console.log('Job sent to delete shard: ', pointer.hash);
```

Comment:
I suppose we want to log every single hash that is sent to be deleted 🤔

Reply:
@sg-gs regarding this. Should we make the networkQueue class have a deleteFile method:

```
deleteFile(hash:string, callback){
    this.enqueueMessage(....)
    callback();
}
```

Inside it instead of setting the type and the payload whenever we use it?

Reply:
As the `networkQueue` enqueues tasks, we could have a `pushFileDeletionTask` method or something similar, that enqueues the deletion task. Could be a good idea. 

Also, take into count the v2 API where the key for the content will not be a hash, changing `hash` for `objectId` (wherever is used as an object identifier, a hash, a uuid, etc) or something like this could be useful.

Reply:
Note: Not an Mongo ObjectId, just a object identifier used to identify uniquely a stored object